### PR TITLE
workflows: avoid unknown/unknown architecture in ghcr

### LINKED ISSUE #2654: Investigate unknown/unknown arch  in the Docker builds
# 🎯 Problem to be solved
An `unknown/unknown` arch is showing up in the Docker builds. It should be related to the lack of version pinning for `buildkit` and `buildx` in the GitHub Actions workflow for `ObolNetwork/charon`. See the discussion [here](https://github.com/orgs/community/discussions/45969).

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBeE51QVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--1e35e38e23a0892c6a5f0262023e3a69d246327a/image.png)

## 🛠️ Proposed solution
- [ ] Investigate the `unknown/unknown` arch issue to confirm that it's related to version pinning of `buildkit` and `buildx`.
- [ ] Update `build-push-deploy.yml` GitHub Actions workflow with version pinning:
  - Option 1: Pin Docker `Buildx` and `buildkit` versions, See [Version Pinning](https://docs.docker.com/build/ci/github-actions/configure-builder/#version-pinning).
    ```yaml
    - name: Set up Docker Buildx
      uses: docker/setup-buildx-action@v3
      with:
        driver-opts: image=moby/buildkit:v0.11.0
    ```
  - Option 2: Add an environment variable to disable default attestations. See [disable the default provenance attestations](https://docs.docker.com/build/building/env-vars/#buildx_no_default_attestations).
    ```yaml
    env:
      BUILDX_NO_DEFAULT_ATTESTATIONS: 1
    ```
We should go for the 2nd option first, since seems less intrusive.

### PR DESCRIPTION
**Pull request by KaloyanTanev**
Pin buildkit version in all `docker/setup-buildx-action@v2` steps (N.B.: at one place v1 is used, which is using buildkit version <0.10.5, so I haven't pinned it there). Other proposed solutions didn't seem to work currently (tested on private repo).

Included as well .vscode in .gitignore.

category: bug
ticket: #2654 

Closes #2654

### CODE REVIEW

**Review by xenowits — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,5 @@
 .idea/
+.vscode/
```
Comment (dB2510) [inline .gitignore:2]: I'd suggest you to shift to goland ide

**Review by dB2510 — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (pinebit) [reply .gitignore:2]: Or, setup global gitignore.

**Review by pinebit — APPROVED** [review]

**Review by pinebit — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (gsora) [reply .gitignore:2]: FWIW whatever IDE one uses it's fine for us.

VSCode is as good (if not better) than Goland, having `.vscode` ignored is okay.

**Review by gsora — APPROVED** [review]

**Comment by gsora** [comment]
Added a "Closes" line to the original PR description, so that #2654 gets closed automagically when this PR is merged!

**Comment by gsora** [comment]
Hey @KaloyanTanev! When you feel comfortable, add the `merge when ready` tag to this PR so that our bot, `obol-bulldozer` can merge it in `main` :D

**Pull request merged**